content/en/docs/tutorials/microservices-3-templating-and-multi-env.md

```diff
@@ -0,0 +1,359 @@
+weight: 3
+---
+
+## Introduction
+The [second tutorial]({{< ref "docs/tutorials/microservices-2-helm-integration" >}}) in this series demonstrated how
+to integrate Helm into your deployment project and how to keep things structured.
+
+The project is however still not flexible enough to be deployed multiple times and/or in different flavors. As an example,
+it doesn't make much sense to deploy redis with replication on a local cluster, as there can't be any high availability
+with single node. Also, the resource requests currently used are quite demanding for a local cluster.
+
+## How to start
+This tutorial is based on the results of the second tutorial. As an alternative, you can take the `2-helm-integration`
+example project found [here](https://github.com/kluctl/kluctl-examples/blob/main/microservices-demo/2-helm-integration)
+and use it the base to be able to continue with this tutorial.
+
+This time, you should start with a fresh kind cluster. If you are sure that you won't loose any critical data by deleting
+the existing cluster, simply run:
+```sh
+$ kind delete cluster
+$ kind create cluster
+```
+
+If you're unsure or if you want to re-use the existing cluster for some reason, you can also simply delete the old deployment:
+```yaml
+$ kluctl delete -t local
+  INFO[0000] Rendering templates and Helm charts
+  INFO[0000] Building kustomize objects
+  INFO[0000] Getting remote objects by commonLabels
+The following objects will be deleted:
+  default/Service/emailservice
+  ...
+  default/ConfigMap/cart-redis-scripts
+  Do you really want to delete 29 objects? (y/N) y
+
+Deleted objects:
+  default/ConfigMap/cart-redis-scripts
+  ...
+  default/StatefulSet/cart-redis-node
+```
+
+The reason to start with a fresh deployment is that we will later switch to different namespaces and stop using the
+`default` namespace.
+
+## Targets
+If we want to allow the deployment to be deployed multiple times, we first need multiple targets in our project. Let's
+add 2 [targets]({{< ref "docs/reference/kluctl-project/targets" >}}) called `test` and `prod`. To do so, modify the
+content of `.kluctl.yml` to contain:
+
+```yaml
+targets:
+  - name: local
+    cluster: kind
+    args:
+      env_type: local
+  - name: test
+    cluster: kind
+    args:
+      env_type: real
+  - name: prod
+    cluster: kind
+    args:
+      env_type: real
+```
+
+You might notice that all targets point to the kind cluster at the moment. This is of course not how you would do it
+in a real project as you'd probably have at least one real production-ready cluster to target your deployments against.
+
+We've also introduced [`args`]({{< ref "docs/reference/kluctl-project/targets#args" >}}) to each target, with one
+actual arguments names `env_type`. This argument will later be used to change details of the deployment, depending
+on the value of it. For example, if it is set to `local`, it doesn't make much sense to have a replicated redis
```

Comment:
For example, if it is set to `local`, it can be used as a toggle for single or multi replica redis deployments.

Reply:
"doesn't make much sense" is used very often

Reply:
Hmm, toggling doesn't sound right in this context. Changed it to:
For example, setting it to `local` might change the redis deployment into a single-node/standalone
deployment.